packages/simulation/pitop/simulation/sprites.py

```diff
@@ -85,3 +102,421 @@ def handle_event(type, target_name, component):
 
         elif type == pygame.MOUSEBUTTONUP:
             component.pin.drive_high()
+
+
+class Buzzer(pygame.sprite.Sprite, SimSprite):
+    Size = (int((PMA_CUBE_SIZE[0] + 2 * MARGIN) * 2), PMA_CUBE_SIZE[0] + 2 * MARGIN)
+
+    def __init__(self, config, scale):
+        pygame.sprite.Sprite.__init__(self)
+
+        def get_pos(image):
+            return (
+                int((self.Size[0] - image.get_width()) / 2),
+                int((self.Size[1] - image.get_height()) / 2),
+            )
+
+        self.component_image = SimSprite._load_image(Images.Buzzer, scale)
+        self.component_pos = get_pos(self.component_image)
+
+        self.buzzer_sound_image = SimSprite._load_image(Images.buzzer_sound_icon, scale)
+        self.buzzer_sound_image_pos = get_pos(self.buzzer_sound_image)
+
+        self.base_image = pygame.Surface(self.Size)
+        self.base_image.fill(pygame.Color("WHITE"))
+        self.rect = self.base_image.get_rect()
+
+    def render(self):
+        value = False
+        if hasattr(self, "state"):
+            value = self.state.get("value", False)
+
+        image = self.base_image.copy()
+        if value:
+            image.blit(self.buzzer_sound_image, self.buzzer_sound_image_pos)
+
+        image.blit(self.component_image, self.component_pos)
+        return image
+
+    def update(self):
+        if not hasattr(self, "state"):
+            return
+
+        self.image = self.render()
+
+
+class LightSensor(pygame.sprite.Sprite, SimSprite):
+    Size = ((PMA_CUBE_SIZE[0] + 2 * MARGIN) * 2, PMA_CUBE_SIZE[1] + 2 * MARGIN)
+
+    def __init__(self, config, scale):
+        pygame.sprite.Sprite.__init__(self)
+
+        self.base_image = self.get_base_image(scale)
+
+        lightbulb_image = SimSprite._load_image(Images.lightbulb_icon, scale)
+        lightbulb_pos = (
+            int(self.Size[0] / 4 * 3 - lightbulb_image.get_width() / 2),
+            MARGIN,
+        )
+        self.base_image.blit(lightbulb_image, lightbulb_pos)
+
+        self.slider = self.get_slider()
+        self.font = pygame.font.SysFont(None, FONT_SIZE)
+        self.text_pos = get_text_position(lightbulb_image, lightbulb_pos)
+
+        if not is_virtual_hardware():
+            self.slider.disable()
+
+        self.image = self.render()
+        self.rect = self.image.get_rect()
+
+    def get_base_image(self, scale):
+        component_image = SimSprite._load_image(Images.LightSensor, scale)
+
+        base_image = pygame.Surface(self.Size)
+        base_image.fill(pygame.Color("WHITE"))
+        base_image.blit(component_image, (MARGIN, MARGIN))
```

Comment:
i suppose even the margins need to be scaled otherwise things wont be positioned correctly relative to the base_image